typing/ctype.ml

```diff
@@ -2319,7 +2319,7 @@ let rec estimate_type_jkind ~expand_component env ty =
   | Tlink _ | Tsubst _ -> assert false
   | Tvariant row ->
      if tvariant_not_immediate row
```

Comment:
Seems odd to have this check outside of `for_boxed_row`.

Reply:
Moved `tvariant_not_immediate` into `Btype` and the check into `Jkind.for_boxed_row`. `tvariant_not_immediate` could also live in `Jkind`, but for now I feel like it fits more into `Btype`. We can move it into `Jkind` when we start considering `void` arguments.